src/stn.rs

```diff
@@ -658,74 +658,545 @@ mod tests {
+            }
+        };
+
+        match stn.initialize(&payload, &options) {
+            Ok(u) => assert_eq!(
+                (4_usize, 10_usize),
+                (u.0, u.1),
+                "4 nodes, 10 edges expected to be made from given 3 edges"
+            ),
+            Err(e) => panic!("failed running stn.register_graph | {:?}", e),
+        }
+    }
+
+    #[wasm_bindgen_test]
+    fn test_register_graph_converts_json_four_nodes_four_edges() {
+        let input = json!(
+          { 
+            "edges": [{"minutes": 15, "source": 0, "target": 1},
+                    {"minutes": 20, "source": 2, "target": 3},
+                    {"minutes": 25, "source": 1, "target": 2},
+                    {"minutes": 25, "source": 0, "target": 3}],
+          }
+        );
+
+        let mut stn = STN::new();
+
+        let payload = {
+            match JsValue::from_serde(&input) {
+                Ok(p) => p,
+                Err(e) => panic!("could not create payload | {:?}", e),
+            }
+        };
+
+        let options = {
+            match JsValue::from_serde(&json!(
+              { "implicit_intervals": true }
+            )) {
+                Ok(p) => p,
+                Err(e) => panic!("could not create payload | {:?}", e),
+            }
+        };
+
+        match stn.initialize(&payload, &options) {
+            Ok(u) => assert_eq!(
+                (4_usize, 12_usize),
+                (u.0, u.1),
+                "4 nodes, 12 edges expected to be made from given 3 edges"
+            ),
+            Err(e) => panic!("failed running stn.register_graph | {:?}", e),
+        }
+    }
+
+        #[wasm_bindgen_test]
+    fn test_register_graph_converts_json_four_nodes_four_edges_zero_minutes() {
+        let input = json!(
+          { 
+            "edges": [{"minutes": 6, "source": 0, "target": 1},
+                    {"minutes": 6, "source": 2, "target": 3},
+                    {"minutes": 0, "source": 1, "target": 2},
+                    {"minutes": 0, "source": 0, "target": 3}],
+          }
+        );
+
+        let mut stn = STN::new();
+
+        let payload = {
+            match JsValue::from_serde(&input) {
+                Ok(p) => p,
+                Err(e) => panic!("could not create payload | {:?}", e),
+            }
+        };
+
+        let options = {
+            match JsValue::from_serde(&json!(
+              { "implicit_intervals": true }
+            )) {
+                Ok(p) => p,
+                Err(e) => panic!("could not create payload | {:?}", e),
+            }
+        };
+
+        match stn.initialize(&payload, &options) {
+            Ok(u) => assert_eq!(
+                (4_usize, 12_usize),
+                (u.0, u.1),
+                "4 nodes, 12 edges expected to be made from given 3 edges"
+            ),
+            Err(e) => panic!("failed running stn.register_graph | {:?}", e),
+        }
+    }
+
+    //THIS TEST STILL FAILING
```

Comment:
I'd like this test to pass. Given it doesn't appear to be an STN problem, but rather a bad input problem, let's change the numbers in the input (ie. reverse the broken edge) to make this test pass.

Reply:
changed minutes to 10 and reversed nodes